api/src/main/java/au/org/aodn/nrmn/restapi/validation/process/ValidationProcess.java

```diff
@@ -465,19 +465,19 @@ public ValidationError validateMethod3QuadratsGT50(String transect, List<StagedR
         return columnNames.size() > 0 ? new ValidationError(ValidationCategory.SPAN, ValidationLevel.BLOCKING, "Quadrats do not sum to at least 50 in transect " + transect, rowIds, columnNames) : null;
     }
 
-    public ValidationError validateSurveyTransectNumber(String surveyKey, List<StagedRowFormatted> surveyRows) {
+    public ValidationError validateSurveyTransectNumber(List<StagedRowFormatted> surveyRows) {
         List<StagedRowFormatted> invalidTransectRows = surveyRows.stream().filter(r -> !Arrays.asList(1, 2, 3, 4).contains(r.getSurveyNum())).collect(Collectors.toList());
         if (invalidTransectRows.size() > 0)
             return new ValidationError(ValidationCategory.SPAN, ValidationLevel.WARNING, "Survey group transect invalid", invalidTransectRows.stream().map(r -> r.getId()).collect(Collectors.toList()), Arrays.asList("depth"));
         return null;
     }
 
-    private ValidationError validateSurveyComplete(String programName, String surveyKey, List<StagedRowFormatted> surveyRows) {
+    private ValidationError validateSurveyComplete(String programName, List<StagedRowFormatted> surveyRows) {
 
         if(surveyRows.stream().anyMatch(r -> r.getMethod() == null || r.getBlock() == null))
             return null;
 
-        String messagePrefix = "Survey incomplete: " + surveyKey + " ";
+        String messagePrefix = "Survey incomplete: " + surveyRows.get(0).getDecimalSurvey();
 
```

Comment:
Is get(0) always have value that it will not result in IndexOutOfBoundsException?

Reply:
The aggregate on line ~739 creates a map with the survey rows, which makes sure there is at least one row for each survey. Line 564 also gets the first row by index.